cmd/system-probe/api/module/factory_linux.go

```diff
@@ -9,12 +9,14 @@ package module
 
 import (
 	sysconfigtypes "github.com/DataDog/datadog-agent/cmd/system-probe/config/types"
+	"github.com/DataDog/datadog-agent/comp/core/workloadmeta"
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
 )
 
 // Factory encapsulates the initialization of a Module
 type Factory struct {
 	Name             sysconfigtypes.ModuleName
 	ConfigNamespaces []string
-	Fn               func(cfg *sysconfigtypes.Config) (Module, error)
+	Fn               func(cfg *sysconfigtypes.Config, wmeta optional.Option[workloadmeta.Component]) (Module, error)
```

Comment:
I don't love how a single dependency of a single module forces all system-probe modules to take it as an argument. That could get out of hand real fast.

Reply:
This is one of the drawback of using this pattern which is widely used in the code base of the Agent. 

It seems in this case it is  quite easily to remove `workloadmeta.Component` from `Fn`. 
In order to do that I can turn [All](https://github.com/DataDog/datadog-agent/blob/273143439781040df3e9e93255e1509e90ba4548/cmd/system-probe/modules/all_linux.go#L18) from a variable to  
```go
func All(wmeta optional.Option[workloadmeta.Component])[]module.Factory{
    // ...
}
```

Note: I can also inject `cfg *sysconfigtypes.Config` into the `All` function in order to make the signature of `Fn` just `func()(Module, error)`
What do you think?

Reply:
I think that would take further rework because `module.Factory` is a type and not an interface. In order to have module-specific dependencies, it would need to be an interface.

Reply:
I don't think the fact that `module.Factory` is a `func` is an issue as you can capture what you need.

```go
// All System Probe modules should register their factories here
func All(cfg *sysconfigtypes.Config, wmeta optional.Option[workloadmeta.Component]) []module.Factory {
	return []module.Factory{
		EventMonitor(cfg, wmeta),
                ....
	}
}

func EventMonitor(cfg *sysconfigtypes.Config, wmeta optional.Option[workloadmeta.Component]) module.Factory {
	// EventMonitor - Event monitor Factory
	return module.Factory{
		Name:             config.EventMonitorModule,
		ConfigNamespaces: eventMonitorModuleConfigNamespaces,
		Fn:               func() (module.Module, error) {
			return createEventMonitorModule(cfg, wmeta),
		},
		NeedsEBPF: func() bool {
			return !coreconfig.SystemProbe.GetBool("runtime_security_config.ebpfless.enabled")
		},
	}
}
```
Note that the signature of `module.Factory.Fn` is `func() (module.Module, error).`